src/Patcher/BsdPatchPatcher.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace cweagans\Composer\Patcher;
+
+use Composer\Util\ProcessExecutor;
+use cweagans\Composer\Patch;
+use Composer\IO\IOInterface;
+
+class BsdPatchPatcher extends PatcherBase
+{
+    protected string $tool = 'patch';
+
+    public function apply(Patch $patch, string $path): bool
+    {
+        // TODO: Dry run first?
+
+        return $this->executeCommand(
+            'patch -p%s --posix --batch -d %s < %s',
+            $patch->depth,
+            $path,
+            $patch->localPath
+        );
+    }
+
+    public function canUse(): bool
+    {
+        $output = "";
+        $result = $this->executor->execute($this->patchTool() . ' --version', $output);
+        // TODO: Is it a valid assumption to assume that if GNU is *not* in the version output, that it's BSD patch?
```

Comment:
For now I think this is ok.  It could be the Apple version of the BSD patch, but in that case it should have at least all of the functionality we have in BSD patch.

Reply:
This may not be a valid assumption when considering the `patch` that comes with Busybox.